From aab6f814a3de00253761f7fb7baa640e87ef924f Mon Sep 17 00:00:00 2001
From: Nicolas Bock <nicolasbock@gmail.com>
Date: Wed, 4 May 2022 10:38:40 -0600
Subject: [PATCH 2/2] More Python 3 fixes

Signed-off-by: Nicolas Bock <nicolasbock@gmail.com>
---
 python/oauth2.py | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/python/oauth2.py b/python/oauth2.py
index 4d69239..28b3f57 100755
--- a/python/oauth2.py
+++ b/python/oauth2.py
@@ -66,7 +66,8 @@ import json
 from optparse import OptionParser
 import smtplib
 import sys
-import urllib
+import urllib.parse
+import urllib.request
 
 
 def SetupOptionParser():
@@ -144,7 +145,7 @@ def AccountsUrl(command):
 
 def UrlEscape(text):
   # See OAUTH 5.1 for a definition of which characters need to be escaped.
-  return urllib.quote(text, safe='~-._')
+  return urllib.parse.quote(text, safe='~-._')
 
 
 def UrlUnescape(text):
@@ -162,7 +163,8 @@ def FormatUrlParams(params):
     A URL query string version of the given parameters.
   """
   param_fragments = []
-  for param in sorted(params.iteritems(), key=lambda x: x[0]):
+  print(params)
+  for param in sorted(params.items(), key=lambda x: x[0]):
     param_fragments.append('%s=%s' % (param[0], UrlEscape(param[1])))
   return '&'.join(param_fragments)
 
@@ -211,7 +213,7 @@ def AuthorizeTokens(client_id, client_secret, authorization_code):
   params['grant_type'] = 'authorization_code'
   request_url = AccountsUrl('o/oauth2/token')
 
-  response = urllib.urlopen(request_url, urllib.urlencode(params)).read()
+  response = urllib.request.urlopen(request_url, urllib.parse.urlencode(params).encode('utf-8')).read()
   return json.loads(response)
 
 
@@ -235,7 +237,7 @@ def RefreshToken(client_id, client_secret, refresh_token):
   params['grant_type'] = 'refresh_token'
   request_url = AccountsUrl('o/oauth2/token')
 
-  response = urllib.urlopen(request_url, urllib.urlencode(params)).read()
+  response = urllib.request.urlopen(request_url, urllib.parse.urlencode(params).encode('utf-8')).read()
   return json.loads(response)
 
 
@@ -321,7 +323,7 @@ def main(argv):
     RequireOptions(options, 'client_id', 'client_secret')
     print('To authorize token, visit this url and follow the directions:')
     print('  %s' % GeneratePermissionUrl(options.client_id, options.scope))
-    authorization_code = raw_input('Enter verification code: ')
+    authorization_code = input('Enter verification code: ')
     response = AuthorizeTokens(options.client_id, options.client_secret,
                                 authorization_code)
     print('Refresh Token: %s' % response['refresh_token'])
-- 
2.34.1

